src/subdomains/staking/application/services/staking.service.ts

```diff
@@ -169,10 +178,10 @@ export class StakingService {
     const withdrawalAddress = await this.userService.getWalletAddress(userId, walletId);
 
     // only one staking per address
-    const existingStaking = await this.repository.findOne({ where: { withdrawalAddress } });
+    const existingStaking = await this.repository.findOne({ where: { withdrawalAddress, strategy: dto.strategy } });
```

Comment:
Also add the asset here? Is not really used right now, but would be "more correct".